public/app/plugins/datasource/cloudwatch/components/MetricStatEditor/MetricStatEditor.test.tsx

```diff
@@ -42,22 +37,22 @@ describe('MetricStatEditor', () => {
       const onRunQuery = jest.fn();
       props.datasource.getVariables = jest.fn().mockReturnValue(['$statistic']);
 
-      render(<MetricStatEditor {...props} onChange={onChange} onRunQuery={onRunQuery} />);
+      render(<MetricStatEditor {...props} refId="a" onChange={onChange} onRunQuery={onRunQuery} />);
 
       const statisticElement = await screen.findByLabelText('Statistic');
       expect(statisticElement).toBeInTheDocument();
 
       await userEvent.type(statisticElement, statistic);
       fireEvent.keyDown(statisticElement, { keyCode: 13 });
-      expect(onChange).toHaveBeenCalledWith({ ...props.query, statistic });
+      expect(onChange).toHaveBeenCalledWith({ ...props.metricStat, statistic });
       expect(onRunQuery).toHaveBeenCalled();
     });
 
     test.each([['CustomStat', 'p23,23', '$statistic']])('should not accept invalid values', async (statistic) => {
       const onChange = jest.fn();
       const onRunQuery = jest.fn();
 
-      render(<MetricStatEditor {...props} onChange={onChange} onRunQuery={onRunQuery} />);
+      render(<MetricStatEditor {...props} refId="a" onChange={onChange} onRunQuery={onRunQuery} />);
```

Comment:
looks like you can put the refId in the `props` object

Reply:
:+1: